src/client/connection.ts

```diff
@@ -57,10 +55,23 @@ export const INTENTIONAL_DISCONNECT_CODE = 4000
 type WebsocketState = 0 | 1 | 2 | 3
 
 function getAgent(url: string, config: ConnectionOptions): Agent | undefined {
-  // TODO: replace deprecated method
   if (config.proxy != null) {
-    const parsedURL = parseURL(url)
-    const parsedProxyURL = parseURL(config.proxy)
+    const parsedURL = new URL(url)
+    const parsedProxyURL = new URL(config.proxy)
+    const proxyInfo = {
+      href: parsedProxyURL.href,
+      origin: parsedProxyURL.origin,
+      protocol: parsedProxyURL.protocol,
+      username: parsedProxyURL.username,
+      password: parsedProxyURL.password,
+      host: parsedProxyURL.host,
+      hostname: parsedProxyURL.hostname,
+      port: parsedProxyURL.port,
+      pathname: parsedProxyURL.pathname,
+      search: parsedProxyURL.search,
+      hash: parsedProxyURL.hash,
+    }
```

Comment:
Instead of this, can we just pass `parsedProxyURL` straight into `proxyOptions` instead of copying it like this?

Reply:
No we cannot. I tried that and because `URL` is a class instance, the `...URL` operator seems to not deconstruct into an object in a clean way.

Reply:
Can we just put those straight into `proxyOptions` instead of having this separate `proxyInfo` object then?

Reply:
Yeah duh, thats a better way to do this.